js/src/forum/addSubscriptionSettings.js

```diff
@@ -2,14 +2,21 @@ import { extend } from 'flarum/extend';
 import SettingsPage from 'flarum/components/SettingsPage';
 import Switch from 'flarum/components/Switch';
 
-export default function() {
-  extend(SettingsPage.prototype, 'notificationsItems', function(items) {
+export default function () {
+  extend(SettingsPage.prototype, 'notificationsItems', function (items) {
     items.add('followAfterReply',
       Switch.component({
-        children: app.translator.trans('flarum-subscriptions.forum.settings.follow_after_reply_label'),
         state: this.user.preferences().followAfterReply,
-        onchange: this.preferenceSaver('followAfterReply')
-      })
+        onchange: (value) => {
+          this.followAfterReplyLoading = true;
+
+          this.user.savePreferences({ followAfterReply: value }).then(() => {
+            this.followAfterReplyLoading = false;
```

Comment:
Do we also need to set `this.followAfterReplyLoading = false` after an error to allow trying again? I've been using a `catch(err => { this.loading = false; throw err; })` kind of syntax in all of my extensions.

The `promise.finally()` function would be nice but I don't think that function is actually standardized and won't work across all browsers.

Reply:
Hmm that's a good point. This is copied from core, which is copied from the old preferenceSaver. Not sure that a change should only be here (or in this commit at all)?

Reply:
I think a good example if what we do in modals. In `onsubmit` we always call `loaded()` on error as well https://github.com/flarum/core/blob/47f3ee0ce2de145de8aec4eb8043d62d4027f000/js/src/forum/components/LogInModal.js#L177

Reply:
Hmm, I'm wondering if it might make sense to bring back something like saveSettings, but instead have it be a util that gets bound to the component... That way we won't need to replicate this as much

Reply:
I don't think we should introduce any new helper for this at this time, however I'd still advocate we handle the catch like in my linked example.

Reply:
That'd still be modifying existing behavior. And would you also want to make the change in saveSettings calls in core?

Reply:
I just did some additional checks and the issue is indeed present in beta13 already with the old `preferenceSaver`, and it's not as bad as I thought since the controls aren't disabled while loading (which I think they should be).

So I created an issue for it and we can tackle that once the Mithril v2 PRs are merged https://github.com/flarum/core/issues/2272